cloudify_cli/commands/blueprints.py

```diff
@@ -96,28 +98,53 @@ def upload(ctx,
     processed_blueprint_path = blueprint.get(
         blueprint_path, blueprint_filename)
 
-    try:
-        if validate:
-            ctx.invoke(
-                validate_blueprint,
-                blueprint_path=processed_blueprint_path)
-        blueprint_id = blueprint_id or blueprint.generate_id(
-            processed_blueprint_path, blueprint_filename)
-
-        progress_handler = utils.generate_progress_handler(blueprint_path, '')
-        logger.info('Uploading blueprint {0}...'.format(blueprint_path))
-        blueprint_obj = client.blueprints.upload(processed_blueprint_path,
-                                                 blueprint_id,
-                                                 progress_handler)
-        logger.info("Blueprint uploaded. The blueprint's id is {0}".format(
-            blueprint_obj.id))
-    finally:
-        # Every situation other than the user providing a path of a local
-        # yaml means a temp folder will be created that should be later
-        # removed.
-        if processed_blueprint_path != blueprint_path:
-            shutil.rmtree(os.path.dirname(os.path.dirname(
-                processed_blueprint_path)))
+    # Take into account that `blueprint.get` might not return a URL
+    # instead of a blueprint file (archive files are not locally downloaded)
+    is_url = bool(urlparse(processed_blueprint_path).scheme)
+
+    progress_handler = utils.generate_progress_handler(blueprint_path, '')
+    blueprint_id = blueprint_id or blueprint.generate_id(
+        processed_blueprint_path, blueprint_filename)
+
+    if is_url:
+        # When a URL is passed it's assumed to be pointing to an archive
+        # file that contains the blueprint. Hence, the `publish_archive`
+        # API call is the one that should be used.
+        logger.info('Publishing blueprint archive %s...', blueprint_path)
```

Comment:
I usually prefer using `.format`, but note that in this case the string interpolation doesn't happen in this line but in the logging code. This is considered to be a good practice in python because if the logging level is set to, let's say, `warning` the interpolation won't happen at all because the message isn't printed.